internal/app/triggerfactory.go

```diff
@@ -61,13 +63,69 @@ func (svc *Service) RegisterCustomTriggerFactory(name string,
 			Logger:           sdk.lc,
 			ContextBuilder:   sdk.defaultTriggerContextBuilder,
 			MessageProcessor: sdk.defaultTriggerMessageProcessor,
+			ProcessMessage:   sdk.processMessageOnRuntime,
 			ConfigLoader:     sdk.defaultConfigLoader,
 		})
 	}
 
 	return nil
 }
 
+func (svc *Service) buildContextForRuntime(envelope types.MessageEnvelope) *appfunction.Context {
+	context := appfunction.NewContext(envelope.CorrelationID, svc.dic, envelope.ContentType)
+
+	if envelope.ReceivedTopic != "" {
+		context.AddValue(interfaces.RECEIVEDTOPIC, envelope.ReceivedTopic)
+	}
+
+	return context
+}
+func (svc *Service) processMessageOnRuntime(envelope types.MessageEnvelope) error {
```

Comment:
```suggestion
func (svc *Service) defaultProcessMessageOnRuntime(envelope types.MessageEnvelope) error {
```

Reply:
correct?? User can override this?

Reply:
I don't see how they would since this is the only public access to the runtime - at that point are you using anything from edgex?  Maybe shouldn't have been before.

Reply:
No need to rename.